domain/modeldefaults/service/service.go

```diff
@@ -39,26 +42,68 @@ type ModelConfigProviderFunc func(string) (environs.ModelConfigProvider, error)
+
+	// GetCloudUUID returns the cloud UUID and region for the given cloud name.
+	// If the cloud is not found, an error specifying [clouderrors.NotFound] is returned.
+	GetCloudUUID(context.Context, string) (cloud.UUID, error)
+
+	// UpdateCloudDefaults is responsible for updating default config values for a
+	// cloud. This function will allow the addition and updating of attributes.
+	// If the cloud doesn't exist, an error satisfying [clouderrors.NotFound]
+	// is returned.
+	UpdateCloudDefaults(ctx context.Context, cloudUID cloud.UUID, attrs map[string]string) error
+
+	// DeleteCloudDefaults deletes the specified cloud default
+	// config values if they exist.
+	DeleteCloudDefaults(ctx context.Context, cloudUID cloud.UUID, attrs []string) error
 
-	// ModelCloud returns the cloud type used by the model identified by the
-	// model uuid. If no model exists for the given uuid then an error
-	// [modelerrors.NotFound] is returned.
-	ModelCloudType(context.Context, coremodel.UUID) (string, error)
+	// UpdateCloudRegionDefaults is responsible for updating default config values
+	// for a cloud region. This function will allow the addition and updating of
+	// attributes. If the cloud is not found an error satisfying [clouderrors.NotFound]
+	// is returned. If the region is not found, am error satisfying [errors.NotFound]
+	// is returned.
+	UpdateCloudRegionDefaults(ctx context.Context, cloudUID cloud.UUID, regionName string, attrs map[string]string) error
+
+	// DeleteCloudRegionDefaults deletes the specified default config
+	// keys for the given cloud region.
+	// It returns an error satisfying [errors.NotFound] if the
+	// region doesn't exist.
+	DeleteCloudRegionDefaults(ctx context.Context, cloudUID cloud.UUID, regionName string, attrs []string) error
 
-	// ModelCloudDefaults returns the defaults associated with the model's cloud.
-	ModelCloudDefaults(context.Context, coremodel.UUID) (map[string]string, error)
+	// ConfigDefaults returns the default configuration values set in Juju.
+	ConfigDefaults(context.Context) map[string]any
 
-	// ModelCloudRegionDefaults returns the defaults associated with the models
-	// set cloud region.
-	ModelCloudRegionDefaults(context.Context, coremodel.UUID) (map[string]string, error)
+	// CloudDefaults returns the defaults associated with the given cloud. If
+	// no defaults are found then an empty map will be returned with a nil error.
+	CloudDefaults(context.Context, cloud.UUID) (map[string]string, error)
+
+	// CloudAllRegionDefaults returns the defaults associated with all of the
+	// regions for the specified cloud. The result is a map of region name
+	// key values, keyed on the name of the region.
+	// If no defaults are found then an empty map will be returned with nil error.
+	// CloudAllRegionDefaults returns all the default settings for a cloud and it's
+	// regions. Note this will not include the defaults set on the cloud itself but
```

Comment:
The one sentence duplicates info in the first sentence of the comment.

```suggestion
	// Note this will not include the defaults set on the cloud itself but
```

Reply:
fixed